modules/core/src/main/scala/Span.scala

```diff
@@ -22,3 +22,18 @@ trait Span[F[_]] {
   def span(name: String): Resource[F, Span[F]]
 
 }
+
+
+sealed trait Propagation extends Product with Serializable
+object Propagation {
+
+  /** Fields will not be propagated to child spans. */
+  case object None     extends Propagation
+
+  /** Fields will be propagated to child spans, but will not cross network boundaries. */
+  case object Local    extends Propagation
+
+  /** Fields will be propagated to child spans, and across network boundaries. */
+  case object Extended extends Propagation
```

Comment:
Do you have any thoughts as to how these Extended params would be made available to things like the http4s middleware so that they can be propagated across the network boundary? I imagine one way is that they could be merged onto the Kernel that the middleware pulls off the trace.

Also, any thoughts about the mechanism the http4s middleware might use to do the propagation? Each field could become a header in the request (perhaps prefixing with `X-Natchez-Field-` to avoid fields being set with names that clash with http headers?) or a single `X-Natches-Fields` header that encodes the name/value pairs into a single string?

One specific thing I'd really like to see, though, is a way to add a field that ends up being set as the `X-Correlation-Id` header on http calls. Maybe that ends up as logic in the middleware to extract a value with a known name from the kernel.

Reply:
Yes, they become part of the kernel. Prefixing with `X-Natchez-Field` is a good idea. We could set the trace ID as the correlation ID although this would need to be optional because it's an existing header people may already be using. I don't really see the point of using both tracing and a correlation ID but I guess people do it.